aiohttp/connector.py

```diff
@@ -170,6 +171,67 @@ def close(self) -> None:
         pass
 
 
+class KeyedCondition:
+    def __init__(self):
+        # NOTE: asyncio.Lock + asyncio.Condition are FIFO
+        self._keyed_cond: Dict[ConnectionKey, asyncio.Condition] = dict()
+
+        # if we wanted to peek into asyncio.Condition._waiters this variable would not be necessary
+        self._num_waiters: Dict[ConnectionKey, int] = dict()
+
+    def keys(self) -> KeysView[ConnectionKey]:
+        return self._keyed_cond.keys()
+
+    async def wait(self, key: ConnectionKey):
+        cond = self._keyed_cond.get(key)
+        if not cond:
+            cond = self._keyed_cond[key] = asyncio.Condition()
+
+        async with cond:
+            try:
+                if key not in self._num_waiters:
+                    self._num_waiters[key] = 1
+                else:
+                    self._num_waiters[key] += 1
```

Comment:
What do you think about making `self._num_waiters` a `defaultdict(int)`?

Reply:
see cmt here: https://github.com/thehesiod/aiohttp/pull/4/files#diff-7f25afde79f309e2f8722c26cf1f10adR202